### PR TITLE
refactor: replace the custom implementation of proxyFromEnvironment with the one provided by golang/net

### PR DESCRIPTION
**Pull request by JeromeTan1997**
### Ⅰ. Describe what this PR did
Remove the custom implementation of parsing proxy from environment variables and adopt the implementation in golang/net

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Describe how you did it
1. Upgrade the vendor golang.org/x/net to commit https://github.com/golang/net/commit/c21de06aaf072cea07f3a65d6970e5c7d8b6cd6d
2. Replace the custom implementation of `proxyFromEnvironment`

### Ⅳ. Describe how to verify it
Run `go test --run ProxyFromEnvironment`

### Ⅴ. Special notes for reviews
To use the `httpproxy`, vendor golang/net has to be upgraded to https://github.com/golang/net/commit/c21de06aaf072cea07f3a65d6970e5c7d8b6cd6d

### CODE REVIEW

File: ctrd/image_proxy_util_test.go [hunk 1]
```diff
@@ -1,82 +1,30 @@
...
-		{name: "test3", urlString: "socks5://shadowsocks.com", want: "shadowsocks.com:1080"},
-		{name: "test4", urlString: "ftp://127.0.0.1", want: "127.0.0.1:"},
-	}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			exampleURL, _ := url.Parse(tt.urlString)
-			got := canonicalAddr(exampleURL)
+func TestProxyFromEnvironment(t *testing.T) {
+	SetImageProxy("http://alibaba.com")
```
Comment (HusterWan) [inline ctrd/image_proxy_util_test.go]: maybe we should not use alibaba.com as example

File: vendor/golang.org/x/net/README [hunk 2]
```diff
@@ -1,3 +0,0 @@
-This repository holds supplementary Go networking libraries.
```
Comment (HusterWan) [inline vendor/golang.org/x/net/README]: why then vendor.json file not changed when updated  the`net` package

File: vendor/vendor.json [hunk 3]
```diff
@@ -1721,10 +1721,10 @@
 			"tree": true
 		},
 		{
-			"checksumSHA1": "HR73EwZ0VzFMLD6RYh4U0OkdaMY=",
+			"checksumSHA1": "EjBv5mhIgEH3MNAGsOEk6Udzqk8=",
 			"path": "golang.org/x/net",
-			"revision": "b3756b4b77d7b13260a0a2ec658753cf48922eac",
-			"revisionTime": "2017-07-16T02:53:09Z",
+			"revision": "c21de06aaf072cea07f3a65d6970e5c7d8b6cd6d",
+			"revisionTime": "2018-05-30T03:23:02Z",
```
Comment (JeromeTan1997) [inline vendor/vendor.json]: @HusterWan Here is it

File: ctrd/image_proxy_util_test.go (hunk 1 above)
Comment (JeromeTan1997) [reply ctrd/image_proxy_util_test.go]: Changed to proxy.example.com

**Comment by fuweid** [comment]
ping @YaoZengzeng could you take a look this fail case? Thanks

```
Summarizing 1 Failure:
[Fail] [k8s.io] Security Context NamespaceOption [It] runtime should support HostPID 
/home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/security_context.go:112
```

**Comment by Ace-Tang** [comment]
@JeromeTan1997 , Could you explain why you do this refactor detailed, we got little information from you comment, since your pr is so large, we need to disscuss about this. Thanks.

**Comment by fuweid** [comment]
seems there is no activity. will reopen if we need it. Thanks

@JeromeTan1997

**Pull request closed without merging**